File: app/src/main/java/de/troido/bledemo/Application.java
package de.troido.bledemo;

import com.karumi.dexter.Dexter;

import java.util.UUID;

public class Application extends android.app.Application {

    public static final UUID UUID = java.util.UUID.randomUUID();

    @Override
    public void onCreate() {
        super.onCreate();
        Dexter.initialize(this);
    }
}

File: app/src/main/java/de/troido/bledemo/Application.kt
package de.troido.bledemo

import android.content.Context
import android.content.SharedPreferences

class Application : android.app.Application() {

    companion object {
        const val PREFERENCES = "preferences"
    }

    override fun onCreate() {
        super.onCreate()
    }
}

val Context.sharedPreferences: SharedPreferences
    get() = getSharedPreferences(
            Application.PREFERENCES,
            Context.MODE_PRIVATE
    )
